test/nearest/rint_test.cpp

```diff
@@ -0,0 +1,408 @@
+/*
+ * Copyright (c) 2024-Present Ian Pike
+ * Copyright (c) 2024-Present ccmath contributors
+ *
+ * This library is provided under the MIT License.
+ * See LICENSE for more information.
+ */
+
+#include <ccmath/math/nearest/rint.hpp>
+
+#include <cmath>
+#include <functional>
+#include <ranges>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+// Disabling test case ccm::nearbyintl if run on clang linux.
+#ifdef __clang__
+	#ifdef __linux__
+		#define CLANG_LINUX
+	#endif
+#endif
+
+namespace
+{
+	using testing::TestWithParam;
+	using testing::ValuesIn;
+
+	template <typename InputType, typename OutputType = InputType>
+	struct RIntTestParams
+	{
+		InputType input{};
+		OutputType output{};
+	};
+
+	template <typename T>
+	std::vector<T> makeTestParams()
```

Comment:
After pushing this all up, I realized I had a question: Is there a specific casing I should be following? I am assuming snake case to match the standard library?

Reply:
There is no hard rule, but generally yes I follow pretty close to LLVMs style using snake case. You can also use clang-format to aid you in making sure your following the formatting style of the project!

Reply:
Sounds good, thanks. I set up the clang format in VSCode to run on every save, however it didn't seem to adjust for the snake vs camel case. I will update the casing for this commit! :)